tiledb/sm/serialization/array.cc

```diff
@@ -134,6 +134,10 @@ Status array_to_capnp(
 
   array_builder->setQueryType(query_type_str(array->get_query_type()));
 
+  if (array->use_refactored_array_open() && array->serialize_enumerations()) {
+    array->load_all_enumerations();
```

Comment:
Please note that in practice, calling `load_all_enumerations` here on the client side will throw:
`throw ArrayException("Unable to load all enumerations; Array is remote.");`
The test passes because the array doesn't look `remote_`.

Reply:
I see that there is C API for `load_all_enumerations`, so I'm guessing the throw is by design, in this case I suggest we load if `!client_side`, on else: if the enumeration map is empty it's fine, it either should've been filled by the deserialization step before with data coming from cloud request or we just don't have enumerations on that array.

Reply:
Good catch here! The missing context is that PR #4288 adds the REST APIs and logic to make this work for clients talking to tiledb-cloud-rest.